### PR TITLE
Add the configuration option to configure static service providers

### PR DESCRIPTION
**Pull request by epinxteren**
(cherry picked from commit d016a5c)

### CODE REVIEW

File: src/Entity/ImmutableCollection/ServiceProviders.php [hunk 1]
```diff
@@ -0,0 +1,62 @@
+<?php
+
+/**
+ * Copyright 2014 SURFnet bv
```
Comment (MKodde) [inline src/Entity/ImmutableCollection/ServiceProviders.php]: ~~2014~~ 2017

File: src/Entity/StaticServiceProviderRepository.php [hunk 2]
```diff
@@ -0,0 +1,43 @@
+<?php
```
Comment (MKodde) [inline src/Entity/StaticServiceProviderRepository.php]: Missing privacy statement

File: src/Entity/StaticServiceProviderRepository.php [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+     */
+    public function __construct(array $serviceProviders)
+    {
+        $this->serviceProviders = new ServiceProviders($serviceProviders);
+    }
+
+    /**
+     * @param string $entityId
+     * @return ServiceProvider
+     * @throws \Surfnet\SamlBundle\Exception\NotFound
```
Comment (MKodde) [inline src/Entity/StaticServiceProviderRepository.php]: Out of curiosity: what is your rationale for using a FQN for the `@throws` but not for the `@return` annotation?

File: src/Exception/SamlInvalidConfigurationException.php [hunk 4]
```diff
@@ -0,0 +1,29 @@
+<?php
+
+/**
+ * Copyright 2014 SURFnet bv
```
Comment (MKodde) [inline src/Exception/SamlInvalidConfigurationException.php]: ~~2014~~ 2017

File: README.md [hunk 5]
```diff
@@ -67,6 +71,10 @@ It is recommended to use parameters as listed above. The various `publickey` and
 The `service_provider_repository` is a repository of service providers for which you offer IdP services. The service
 configured _must_ implement the `Surfnet\SamlBundle\Entity\ServiceProviderRepository` interface.
 
+Static remote service providers can be configured in `remote.service_providers`. Keep in mind that you need to assign
```
Comment (MKodde) [inline README.md]: It's not really clear to me what can be configured. Maybe it's a good idea to clarify what a Static remote service provider is. And why a potential user might be interested in using it.

File: src/Entity/ImmutableCollection/ServiceProviders.php (hunk 1 above)
Comment (epinxteren) [reply src/Entity/ImmutableCollection/ServiceProviders.php]: The current header is based on 2014, then i need to change all files

File: README.md [hunk 6]
```diff
@@ -67,6 +71,9 @@ It is recommended to use parameters as listed above. The various `publickey` and
 The `service_provider_repository` is a repository of service providers for which you offer IdP services. The service
 configured _must_ implement the `Surfnet\SamlBundle\Entity\ServiceProviderRepository` interface.
 
+Service providers can statically be provided by configuration `remote.service_providers`. To use these configured service
```
Comment (MKodde) [inline README.md]: Suggestion to improve grammer:

Original:
| Service providers can statically be provided by configuration `remote.service_providers`.

Suggestion:
| Service providers can be provided statically by using the `remote.service_providers` configuration option.

**Pull request merged**